### ISSUE TITLE
Implicit exists() or not?

### ISSUE THREAD

**Issue by ijohn** [ctavan/express-validator#399]
I am in the process of migrating my express-validator 3.2.1 app to the new v4 API (4.1.0). In order to make my question easier, I already setup a test app.

I am using Node.js 8.4.0 with npm 5.3.0 BTW.

Please run the following:
```
git clone git@github.com:ijohn/test-validator.git
cd test-validator
npm i
npm start
```

Test using `curl`:
```
curl http://localhost:3000/\?email\=one%40example.com
```

The following is the result:
```
{"email":"one@example.com"}
```

What I expected:
```
{"errors":{"name":{"location":"query","param":"name","msg":"must be ASCII"}}}
```

Question is, why does
`query('email').isEmail().withMessage('must be valid')` gets an implicit `exists()` while `query('name').isAscii().withMessage('must be ASCII')` doesn't?

**Comment by gustavohenke** [comment]
This seems to be a bug, actually.

See the relevant lines in `runner.js`:
https://github.com/ctavan/express-validator/blob/8e3de682703b48d64629a5469b43309081b6bfe2/check/runner.js#L8-L10

Since validator only accepts strings, values passed to the standard validators get converted to string.
What means `isAscii` is being called with `"undefined"` (a string) internally; therefore, valid value.

This behaviour must be changed (perhaps use the same as v3), and possibly include a comprehensive integration test suite, testing each validator function 🛠 👨‍🏭

**Comment by ijohn** [comment]
Ah! So the reason `isEmail()` seems to have implicit `exists()`, is because `undefined` is not a valid email address.

Thus for `4.1.0`, we have to add `exists()` if it's not `optional()`.

**Comment by gustavohenke** [comment]
*Yes*, but please don't take this for granted. It's a bug.

**Comment by gustavohenke** [comment]
Fix published in v4.1.1.
